src/openfl/_internal/renderer/RenderSession.hx

```diff
@@ -58,6 +63,37 @@ class RenderSession {
 	//public var defaultFramebuffer:GLFramebuffer;
 	
 	
+	private function set_gl(gl: GLRenderContext): GLRenderContext {
+		
+		this.gl = gl;
+		
+		#if vertex_array_object
+		if (gl.type == GLContextType.WEBGL) { 
+			
+			if (gl.version == 2) {
+				
+				vaoContext = new VertexArrayObjectContext (gl);
```

Comment:
I think OpenFL is not 100% consistent when it comes to "this" usage on setting members, but because "this.gl" is used already, maybe a consistent way inside this function make sense